File: src/main/scala/leetcode/Solution268.scala
package leetcode

object Solution268 extends App {

  /**
   * Given an array nums containing n distinct numbers in the range [0, n],
   * return the only number in the range that is missing from the array.
   * @param nums
   * @return
   */
  def missingNumber(nums: Array[Int]): Int = {
    val newNums = for(i <- 0 to nums.length) yield i
    newNums.dropWhile(i => nums.contains(i)).head
  }


}

File: java-solution/src/main/java/leetcode/Solution268.java
package leetcode;

import java.util.ArrayList;
import java.util.List;

public class Solution268 {

    public int missingNumber(int[] nums) {
        List<Integer> numsToList = new ArrayList<>(nums.length);
        List<Integer> newNumToList = new ArrayList<>(nums.length + 1);
        for(int i = 0; i < nums.length; i++) numsToList.add(nums[i]);
        for(int i = 0; i <= nums.length; i++) newNumToList.add(i);
        return newNumToList.stream().dropWhile(numsToList::contains).findFirst().get();
    }

}
